PowerShell/CipherTrustManager/Connections/CipherTrustManager-Connections-IdP.psm1

```diff
@@ -0,0 +1,1465 @@
+        "ConvertTo-JSON:Depth"=5
+    }
+}else{
+    Write-Debug "Setting PS5.1 Defaults - Connections IdP Module"
+    $PSDefaultParameterValues = @{"ConvertTo-JSON:Depth"=5}
+    # Allow the use of self signed certificates and set TLS
+    [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+    # C# class to create callback
+    $code = @"
+    public class SSLHandler
+    {
+        public static System.Net.Security.RemoteCertificateValidationCallback GetSSLHandler()
+        {
+            return new System.Net.Security.RemoteCertificateValidationCallback((sender, certificate, chain, policyErrors) => { return true; });
+        }
+    }
+"@
+    # Compile the class
+    Add-Type -TypeDefinition $code
+
+    #disable checks using new class
+    [System.Net.ServicePointManager]::ServerCertificateValidationCallback = [SSLHandler]::GetSSLHandler()
+}
+
+
+#This project mirrors the "Connections: Connections" section of the API Playground of CM (/playground_v2/api/Connection Manager/Connections) for the creation of Identity Providers for CM Authentication
+
+#Connection Manager - Connections (IdP)
+#"#/v1/usermgmt/connections/"
+#"#/v1/usermgmt/connections/ - get"
+
+<#
+    .SYNOPSIS
+        List all CipherTrust Manager IdP Connections
+    .DESCRIPTION
+        Returns a list of all connections. The results can be filtered using the query parameters.
+        Results are returned in pages. Each page of results includes the total results found, and information for requesting the next page of results, using the skip and limit query parameters. 
+        For additional information on query parameters consult the API Playground (https://<CM_Appliance>/playground_v2/api/Connection Manager/Connections).   
+    .PARAMETER skip
+        The index of the first resource to return. Equivalent to `offset` in SQL.
+    .PARAMETER limit
+        The max number of resources to return. Equivalent to `limit` in SQL.
+    .PARAMETER sort
+        The field, or fields, to order the results by. This should be a comma-delimited list of properties.
+        For example, "name,-createdAt" .. will sort the results first by 'name', ascending, then by 'createdAt', descending.
+    .PARAMETER strategy
+        Search for connection Identity Provider type:
+            -ldap
+            -oidc
+    .EXAMPLE
+        PS> Find-CMIdPConnections -strategy ldap
+        Returns a list of all LDAP Identity Providers
+    .LINK
+        https://github.com/thalescpl-io/CDSP_Orchestration/tree/main/PowerShell/CipherTrustManager
+#>
+function Find-CMIdPConnections {
+    param
+    (
+        [Parameter()] [int] $skip,
+        [Parameter()] [int] $limit,
+        [Parameter()] [string] $sort,
+        [Parameter()] [ValidateSet("ldap","oidc")] [string] $strategy
+    )
+    Write-Debug "Start: $($MyInvocation.MyCommand.Name)"
+    
+    Write-Debug "Getting a List of all LDAP Connections in CM"
+    $endpoint = $CM_Session.REST_URL + $target_uri
```

Comment:
$target_uri is set to "/usermgmt/connections"... 
there is also an API for listing all external identity providers /auth/id-providers

I am assuming in this module we are just referring to OIDC/LDAP based identities associated with user management tile.

Reply:
Feels like the /autg/id-prividers section oft he playground only has to do with actually authenticating people. not "managing" the connections themselves.